gamla/functional.py

```diff
@@ -186,37 +186,17 @@ def compute_stable_json_hash(item) -> Text:
     ).hexdigest()
 
 
-def star(function: Callable) -> Callable:
-    """Turns a variadic function into an unary one that gets a tuple of args to the original function.
-
-    >>> from gamla import functional_generic
-    >>> functional_generic.pipe((2, 3), star(lambda x, y: x + y))
-    5
-    """
-
-    def star_and_run(x):
-        return function(*x)
-
-    async def star_and_run_async(x):
-        return await function(*x)
-
-    if inspect.iscoroutinefunction(function):
-        return star_and_run_async
-    return star_and_run
-
-
-@currying.curry
-def _assert_f_output_on_inp(f, inp):
-    assert f(inp)
-
-
 def assert_that(f):
     """Assert a function `f` on the input.
 
     >>> assert_that(equals(2))(2)
     2
     """
-    return curried.do(_assert_f_output_on_inp(f))
+
+    def assert_that_f(inp):
+        assert f(inp)
+
+    return curried.do(assert_that_f)
```

Comment:
I think you wanted to return `assert_that_f` and avoid the curried.do

Reply:
hmm i think it's good, assert_that should have a `do` (not interfere with the next function in the pipeline)